src/Plugin/GraphQL/Schema/SdlSchemaPluginBase.php

```diff
@@ -179,7 +193,7 @@ protected function getExtensions() {
    */
   protected function getSchemaDocument(array $extensions = []) {
     // Only use caching of the parsed document if we aren't in development mode.
-    $cid = "schema:{$this->getPluginId()}";
+    $cid = "server:{$this->serverId}:schema:{$this->getPluginId()}";
```

Comment:
This assumes that the setter has always been called, but it might not have been, and `$this->serverId` might be `NULL`.

Probably using a setter is not the best approach. Since the server ID is required for the plugin to work in the current it is essential that the setter is always called. The setter is not part of the `SchemaPluginInterface` and it also should not be, since it is not essential for all schema plugins, only for the ones that use caching.

Maybe we can add a `server_name` key to the standard plugin configuration and check that this is populated in the plugin constructor? That would also enable us to gracefully introduce this change. We can emit a deprecation warning if it is missing.